pkg/util/push/push.go

```diff
@@ -46,14 +47,15 @@ func Handler(
 	sourceIPs *middleware.SourceIPExtractor,
 	allowSkipLabelNameValidation bool,
 	push Func,
+	replicaChecker distributor.HaShortcutRequestCheckerFunc,
```

Comment:
There's no need to introduce a dependency on the distributor here, not being specific of what kind of shortcut this is, so let's do:

```suggestion
	shortcut func(context.Context, http.ResponseWriter, *http.Request) bool,
```